include/openthread/border_routing.h

```diff
@@ -326,6 +326,59 @@ otError otBorderRoutingGetNextPrefixTableEntry(otInstance
                                                otBorderRoutingPrefixTableIterator *aIterator,
                                                otBorderRoutingPrefixTableEntry    *aEntry);
 
+/**
+ * Adds an on mesh prefix from an ICMPv6 RouterAdvertisement message.
+ *
+ * Only the first PIO in the most recent RA message will be applied to the thread network data.
+ *
+ * The desired use case is the prefix will be allocated by other softwares on the interface, and they will advertise the
+ * assigned prefix to the thread interface via router advertisement messages.
+ *
+ * `OPENTHREAD_CONFIG_BORDER_ROUTING_ACCEPT_PLATFORM_RA_ENABLE` must be enabled.
+ *
+ * @param[in] aInstance A pointer to an OpenThread instance.
+ * @param[in] aMessage  A pointer to an ICMPv6 RouterAdvertisement message.
+ * @param[in] aLength   The length of ICMPv6 RouterAdvertisement message.
+ *
+ * @retval OT_ERROR_NONE          Successfully processed the prefix infomation in the message.
+ * @retval OT_ERROR_INVALID_STATE Routing manager is configured to not handling RA.
+ *
+ */
+otError otBorderRoutingAddPrefixByRouterAdvertisement(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
```

Comment:
Can we instead use the `otPlatInfraIfRecvIcmp6Nd()` to pass the RA message back to OT?
- We can use `aSrcAddress` such that it is the device's address.
- Then in `RoutingManager` we can tell if this RA is generated by platform by checking the address using `otPlatInfraIfHasAddress()` (and process it differently)?

This way we don't need to add new platform API.

Reply:
@abtink RA in `otPlatInfraIfRecvIcmp6Nd()` is message received on the infra interface, but the RA here is received on the Thread tunnel interface. I don't think it's going to work

Reply:
I see. Make sense. Thanks.